etna/search/forms.py

```diff
@@ -234,8 +248,26 @@ def clean(self):
                 "opening_end_date"
             ):
                 self.add_error(
-                    "opening_start_date",
-                    CUSTOM_ERROR_MESSAGES.get("invalid_date_range"),
+                    None,
+                    ValidationError(
+                        _(CUSTOM_ERROR_MESSAGES.get("invalid_date_range")),
+                        code="invalid_date_range_for_opening_dates",
```

Comment:
Using `invalid_date_range` as the code for both sets of dates should be fine. Think of it more as a 'category' for the error. You already know the 'for_opening_dates' part because of the field to which the error is attached.

Reply:
Done. Thanks.

Reply:
Further to comment below -
> @ababic , I've update the comments in the ticket for the non field error. The error is not specific to a particular field, but a set of fields. I've changed to reflect the same code as `invalid_date_range` but I know think it should be made separate if at some point we want to shift errors to another location where it can be distinguished for with set of fields its related to.

Should I revert this back to individual codes here?

Reply:
@ababic 
1) The want is to **keep the existing functionality** i.e. these date errors - specifically when start and end date are entered together must appear above the respective set date fields - Created dates and Record opening dates - see ticket comment. Unfortunately the existing functionality induces a **non-field error** into a field - ex `record_opening_start_date` which I sought to update as we now have 2 sets of dates.
2) How would we then now which non field error is for which set of the date fields? I have removed the identification generated from the tag.
3) About error code in `add_error` method - I got some details for here when I did it.
https://docs.djangoproject.com/en/4.2/ref/forms/api/#django.forms.Form.add_error
> The error argument can be a string, or preferably an instance of ValidationError

3) Do you want to discuss another structure, please bring it to the ticket and concerned?